app/src/sharedTest/java/org/oppia/app/home/HomeActivityTest.kt

```diff
@@ -185,34 +205,47 @@ class HomeActivityTest {
     launch(HomeActivity::class.java).use {
       onView(withId(R.id.home_recycler_view)).perform(scrollToPosition<RecyclerView.ViewHolder>(1))
       onView(isRoot()).perform(orientationLandscape())
-      onView(atPositionOnView(R.id.home_recycler_view, 1, R.id.story_name_text_view)).check(
-        matches(
-          withText(containsString("Second Story"))
+      onView(
+        Matchers.allOf(
+          withId(R.id.promoted_story_list_recycler_view),
+          ViewMatchers.withParent(
+            atPosition(R.id.home_recycler_view, 1)
+          )
         )
-      )
+      ).check(matches(ViewMatchers.hasDescendant(withText(Matchers.containsString("The Meaning of Equal Parts")))))
     }
   }
 
   @Test
   fun testHomeActivity_recyclerViewIndex1_clickPromotedStory_opensTopicActivity() {
     launch(HomeActivity::class.java).use {
       onView(withId(R.id.home_recycler_view)).perform(scrollToPosition<RecyclerView.ViewHolder>(1))
-      onView(atPosition(R.id.home_recycler_view, 1)).perform(click())
+      onView(
+        Matchers.allOf(
+          withId(R.id.promoted_story_list_recycler_view),
+          ViewMatchers.withParent(
+            atPosition(R.id.home_recycler_view, 1)
+          )
+        )
+      ).perform(click())
       intended(hasComponent(TopicActivity::class.java.name))
-      intended(hasExtra(TopicActivity.TOPIC_ACTIVITY_TOPIC_ID_ARGUMENT_KEY, "test_topic_id_0"))
-      intended(hasExtra(TopicActivity.TOPIC_ACTIVITY_STORY_ID_ARGUMENT_KEY, "test_story_id_1"))
+      intended(hasExtra(TopicActivity.TOPIC_ACTIVITY_TOPIC_ID_ARGUMENT_KEY, "GJ2rLXRKD5hw"))
```

Comment:
Can we pull in the corresponding constants from the domain layer controllers instead of hardcoding the IDs?

Reply:
Done, all IDs in this file are now fetched from domain layer.